diff/apply/apply.go

```diff
@@ -88,8 +89,12 @@ func (s *fsApplier) Apply(ctx context.Context, desc ocispec.Descriptor, mounts [
 	defer processor.Close()
 
 	digester := digest.Canonical.Digester()
+	reader := ioutil.NewConReader(processor)
+	defer func() {
+		reader.Close()
+	}()
 	rc := &readCounter{
-		r: io.TeeReader(processor, digester.Hash()),
+		r: io.TeeReader(reader, digester.Hash()),
```

Comment:
oh so if this makes things faster, then it means that digester.Hash.Write (calculating the hash) is the bottleneck here.